envisage/tests/test_extension_point_changed.py

```diff
@@ -540,8 +622,10 @@ def test_race_condition(self):
         # Now we start the application, which connects the listener.
         application.start()
 
-        # then
-        self.assertEqual(a.x, [4, 5, 6, 98, 99, 100])
+        # fixme: If the extension point has not been accessed then the
+        # provider extension registry can't work out what has changed, so it
+        # won't fire a changed event.
+        self.assertEqual([1, 2, 3, 98, 99, 100], a.x)
```

Comment:
This demonstrates the race condition in this comment:

https://github.com/enthought/envisage/blob/bae1be501d9c5c18a1cffdc7d681ef0102af413d/envisage/extension_point.py#L186-L190